mantle/kola/tests/ignition/kdump.go

```diff
@@ -0,0 +1,205 @@
+		     "sshAuthorizedKeys":["%s"]
+			}
+		  ]
+       }
+    }`, strings.TrimSpace(string(pubkeyBuf))))
+
+
+    // get machine address
+	switch pc := c.Cluster.(type) {
+		// These cases have to be separated because when put together to the same case statement
+		// the golang compiler no longer checks that the individual types in the case have the
+		// NewMachineWithQemuOptions function, but rather whether platform.Cluster
+		// does which fails
+		case *qemu.Cluster:
+			m, err = pc.NewMachineWithQemuOptions(ignition, options)
+			for _, hfp := range options.HostForwardPorts {
+				if hfp.Service == "ssh" {
+					address = fmt.Sprintf("10.0.2.2")
+					port = fmt.Sprintf("%d",hfp.HostPort)
+				}
+			}
+		default:
+			m, err = pc.NewMachine(ignition)
+			address = m.IP()
+			port = "22"
+		}
+		if err != nil {
+			c.Fatal(err)
+		}
+
+	// FIXME this is a bit ugly
+    // insert indentation in front of the entries
+	// to avoid errors in the butane file
+	var padded = ""
+	for _, line := range strings.Split(strings.TrimSuffix(string(PrivKeyBuf), "\n"), "\n") {
+		padded = fmt.Sprintf("%s          %s\n", padded, line)
+	}
+
+	return SshServer {
+		Machine:    m,
+		MachineAddress:    address,
+	    SSHPort: port,
+		PubSSH: string(pubkeyBuf),
+		PrivSSH: padded,
+	}
+}
+
+func kdumpSSHTest(c cluster.TestCluster) {
+	ssh_host := setupSSHMachine(c)
+	butane := conf.Butane(fmt.Sprintf(`variant: fcos
+version: 1.5.0
+storage:
+  files:
+    - path: /root/.ssh/id_ssh_kdump.pub
+      mode: 0600
+      contents:
+        inline: |
+          %s
+    - path: /root/.ssh/id_ssh_kdump
+      mode: 0600
+      contents:
+        inline: |
+%s
+    - path: /root/.ssh/config
+      mode: 0644
+      overwrite: true
+      contents:
+        inline: |
+           Host %s
+               StrictHostKeyChecking no
+               Port %s
+    - path: /etc/kdump.conf
+      overwrite: true
+      contents:
+        inline: |
+          ssh core@%s
+          sshkey /root/.ssh/id_ssh_kdump
+          path /home/core/crash
+          core_collector makedumpfile -F -l --message-level 1 -d 31
+systemd:
+  units:
+    - name: kdump.service
+      enabled: true
+kernel_arguments:
+    should_exist:
+      - crashkernel=512M`,
+	   ssh_host.PubSSH, ssh_host.PrivSSH, ssh_host.MachineAddress, ssh_host.SSHPort, ssh_host.MachineAddress))
+
+	opts := platform.MachineOptions{
+		MinMemory: 2048,
+	}
+
+	kdump_machine, err := c.NewMachineWithOptions(butane, opts)
+	if err != nil {
+		c.Fatalf("Unable to create test machine: %v", err)
+	}
+
+	// TODO : maybe add a retry loop +  timeout ?
+    time.Sleep(15 * time.Second)
+
+	// make sure kdump service is active
+	kdump_status := c.MustSSH(kdump_machine, "systemctl is-active kdump.service")
+	if string(kdump_status) == "inactive" {
+       c.Fatalf("Kdump.service is not ready: %s. Err: %v", string(kdump_status), err)
+	}
+
+	// crash the kernel
+	c.SSH(kdump_machine, "sudo su -c \"echo 'c' > /proc/sysrq-trigger\"")
```

Comment:
Looking at the GitHub Actions error, one thing comes to my mind here:
if err != nil {
		c.Fatalf("Unable to execute: %v", err)
	}

Reply:
This command will always hangs as it crashes the kernel so it will always errors out. 
Added a comment and explicitly ignored the error